### DISCUSSION TITLE
Proposal to change the output of swanctl --list-pools command

### DISCUSSION THREAD

**Discussion by harisasank** [strongswan/strongswan#2263] [Ideas]
swanctl --list-pools displays 'base' address and 'size' of the pool.  Intuatively, one feels that 'base' address is the first available address belonging to that pool. However, that doesn't seem to be the case.
For Ex: 
If `/etc/ike/swanctl/conf.d/pools.conf` is configured as: 
 ```
p1 {
    addrs = 192.168.1.0-192.168.1.253
}
p2 {
    addrs = 192.168.0.0/24
}
```
The output of  **swanctl --list-pools --raw** is as follows
```get-pools reply {p1 {base=192.168.1.0 size=254 online=0 offline=0} p2 {base=192.168.0.0 size=254 online=0 offline=0}```

I propose changing the base address in the command to display the first available address like -
```get-pools reply {p1 {base=192.168.1.0 size=254 online=0 offline=0} p2 {base=192.168.0.1 size=254 online=0 offline=0}```
192.168.0.1 because when pools are configured in cidr form, we ignore the network address for leases. 

I can submit a pull request to implement this if everybody else thinks that this would be meaningful. 
Thank you!

**Comment by tobiasbrunner** [comment]
> because when pools are configured in cidr form, we ignore the network address for leases.

If the base is the network ID of the subnet, then an internal offset is applied and the size of the pool is adjusted, the base remains as configured. But if you configure an offset, e.g. `addrs = 192.168.0.2/24` (or even `192.168.0.1/24`), then the first address is actually the base. I guess instead of just adding the offset internally in the first case, we could adjust the base address to represent this properly. A minor caveat is that it wouldn't be possible to determine how the pool was configured (whether it was e.g. `192.168.0.0/24` or `192.168.0.1/24`). But since it's also not possible to differentiate between pools that were created from ranges or from CIDR notation (unless there is clearly no CIDR representation for a particular base and size), that might not be much of an issue. See the _2263-mem-pool-base_ branch for a possible change.

**Reply by harisasank** [reply]
Thanks Tobias. This patch addresses my concern.
